docs/refguide/autorun.md

```diff
@@ -1,110 +1,355 @@
+Applications without side effects are meaningless; if an application doesn't cause a physical change in the world somewhere, it even doesn't have to run after all.
+So the significance of reactions is to create consumers for your observable state and _automatically_ run side effects whenever something _relevant_ changes.
+
+However, with that in mind, it is important to realize that the APIs discussed here should rarely be used.
+Typically they are abstracted away in other libraries (like mobx-react) or abstractions that are specific to your application.
+
+But, to grok MobX, let's take a look at how reactions can be created.
+The simplest way is to use the [`autorun`](#autorun) utility.
+Beyond that, there is also [`reaction`](#reaction) and [`when`](#when).
+
+## Autorun
+
+Usage:
+
+-   `autorun(effect: (reaction) => void)`
+
+The `autorun` function accepts one function that should run each time anything it observes changes.
+It also runs once when you create the `autorun`. It only responds to changes in observable state -- so things you marked `observable` or `computed`.
 
-`mobx.autorun` can be used in those cases where you want to create a reactive function that will never have observers itself.
-This is usually the case when you need to bridge from reactive to imperative code, for example for logging, persistence, or UI-updating code.
-When `autorun` is used, the provided function will always be triggered once immediately and then again each time one of its dependencies changes.
-In contrast, `computed(function)` creates functions that only re-evaluate if it has
-observers on its own, otherwise its value is considered to be irrelevant.
-As a rule of thumb: use `autorun` if you have a function that should run automatically but that doesn't result in a new value.
-Use `computed` for everything else. Autoruns are about initiating _effects_, not about producing new values.
-If a string is passed as first argument to `autorun`, it will be used as debug name.
+### How tracking works
 
-The return value from autorun is a disposer function, which can be used to dispose of the autorun when you no longer need it. The reaction itself will also be passed as the only argument to the function given to autorun, which allows you to manipulate it from within the autorun function. This means there are two ways you can dispose of the reaction when you no longer need it:
+Autorun works by running `effect` in a _reactive context_; during the executing of the provided function MobX will keep track of all observable and computed values that are directly or indirectly _read_ by the effect.
+Once the function finishes, MobX will collect and subscribe to all observables that were read and wait until any of them changes again.
+Once any of these observables changes, the autorun will trigger again, repeating the entire process.
+In a picture, the code-example below works like this:
+
+![autorun](../assets/autorun.png)
+
+### Example
```

Comment:
nit/style: `reaction` and `when` wrap their examples in `<details>`, but this one does not. Unsure if its intentional, but I find the wrapped ones a bit more pleasant